CMakeLists.txt

```diff
@@ -70,14 +70,48 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 message("-- C++ Standard version: ${CMAKE_CXX_STANDARD}")
 
 if ("CUDA" STREQUAL "${THRUST_DEVICE_SYSTEM}")
-  if (NOT "${CMAKE_CUDA_HOST_COMPILER}" STREQUAL "")
-    unset(CMAKE_CUDA_HOST_COMPILER CACHE)
-    message(FATAL_ERROR "Thrust tests and examples require the C++ compiler"
-        " and the CUDA host compiler to be the same; to set this compiler, please"
-        " use the CMAKE_CXX_COMPILER variable, not the CMAKE_CUDA_HOST_COMPILER"
-        " variable.")
+  if ("Feta" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
+    if (NOT "${CMAKE_CXX_COMPILER}" STREQUAL "")
+      unset(CMAKE_CXX_COMPILER CACHE)
+      message(FATAL_ERROR "You are using Feta as your CUDA C++ compiler, but have"
+          " specified a different ISO C++ compiler; Feta acts as both, so please"
+          " unset the CMAKE_CXX_COMPILER variable.")
+    endif ()
+    if (NOT "${CMAKE_CUDA_HOST_COMPILER}" STREQUAL "")
+      unset(CMAKE_CUDA_HOST_COMPILER CACHE)
+      message(FATAL_ERROR "You are using Feta as your CUDA C++ compiler, but have"
+          " specified a different host ISO C++ compiler; Feta acts as both, so"
+          " please unset the CMAKE_CUDA_HOST_COMPILER variable.")
+    endif ()
+    set(CMAKE_CXX_COMPILER ${CMAKE_CUDA_COMPILER})
+  endif () 
+
+  enable_language(CXX)
```

Comment:
Why not just have the user specify `CMAKE_CXX_COMPILER` correctly when they configure? Changing the compiler variables like this is a really odd pattern for CMake, typically these would be specified to cmake:

```
cmake -DCMAKE_CUDA_COMPILER="...." -DCMAKE_CXX_COMPILER="..." /path/to/src
```

Out of curiosity, can Feta cuda objects not be mixed with other c++ compilers?

Reply:
I'd rather just see an error message if we need to enforce these to be the same. The `*_COMPILER` variables are special to CMake -- if they change, cmake will freak out, drop all custom configuration option, and reconfigure from scratch. That behavior makes me nervous about trying to be too clever here.

Reply:
No, you can't mix cheeses. Feta objects are not compatible with Cheddar (aka NVCC) objects or GoUDA objects.

Reply:
Feta really ought to be both the CUDA compiler and the C++ compiler, at least when compiling Thrust.